### ISSUE TITLE
Passkey Support

### ISSUE THREAD

**Issue by dtantono** [buttercup/buttercup-core#334]
Hi
Many password manager want to support passkey nowadays. Does buttercup also want to support passkey?

**Comment by perry-mitchell** [comment]
Hello! Yes, I can say that Buttercup will eventually support several other kinds of authentication such as passkeys. 

I would add it myself - passkey support - but I'm currently tied up with other tasks. I'd love to see some PRs in support of this feature.. though to be honest I haven't read much into them nor perhaps fully grasp what they're capable of. I'd also appreciate more information on how such an integration would work 😊.

**Comment by ldexterldesign** [comment]
👋

[This] was/is my intro' to [passkey] - I should probably re-listen before I make it a recommendation or not though 😬

Hope it helps!

Regards

[passkey]: https://en.wikipedia.org/wiki/WebAuthn
[this]: https://syntax.fm/show/710/a-passwordless-future-passkeys-with-anna-pobletts

**Comment by ldexterldesign** [comment]
👋

I'm back 🙂

Hope y'all well

I just spent an evening researching this and TBH it's already an evening too much 😮‍💨

TLDR (😒😮‍💨) github.com (GH) is now enforcing MFA and want to keep using GH so I see options to setup SMS or authenticator app. My first thought is "what if I lose my smartphone?" - SMS will be catastrophic because I'll lose my SIM so that leaves authenticator app. I've used authy in the past and deleted it (for some reason) then I read [this] horror story which tells me it was for some *good* reason. So now I'm not that riveted to use another authenticator app but if I lost my smartphone and had some authenticator app backup key saved in BC I could restore everything and get access to everything without my smartphone, which appeals. Enter passkeys, which seem to be the authenticator app replacement/upgrade. I set-up everything in new Apple Passwords app login to GH using a passkey and realise all this is locked to Apple (i.e. no way to export anything from iCloud, no way to access anything on a non-Apple device). So this leads me to go back to SMS with the pending problem of finding a platform agnostic solution to MFA and/or passkeys - best thing I can think to do at this point is find a device/platform agnostic authenticator app and save the backup key for that in BC.

@perry-mitchell [that] is useful!

Where are others with this dilemma?

Hope to hear back

Cheers

[that]: https://github.com/buttercup/buttercup-core/issues/294
[this]: https://news.ycombinator.com/item?id=40878590

**Comment by ldexterldesign** [comment]
For time being settled on Bitwarden Authenticator ([BA]) to cover the 2FA side of things, until a suitable passkey solution comes up - better than SMS I suppose 🫤

Hope this helps!

[BA]: https://alternativeto.net/software/bitwarden-authenticator/about/

**Comment by zicklag** [comment]
In case this wasn't already clear, Buttercup can work as an authenticator app, generating those TOTP ( time-based one time password ) codes. I use it to login to GitHub.
